packages/components/src/app/word-query/word-query-adapter.js

```diff
@@ -0,0 +1,115 @@
+import { ApolloClient, InMemoryCache, HttpLink, gql } from '@apollo/client/core'
+
+let singleInstance
+
+const WORD_QUERY =
+  gql`
+    query WordQuery($langCode: Language, $userID: String) {
+      word(langCode: $langCode, userID: $userID) @client
+    }
+  `
+
+/**
+ * Converts a results of GraphQL word query into JS objects.
+ * Provides several methods to retrieve data via GraphQL word query.
+ */
+export default class WordQueryAdapter {
+  constructor ({ wordQueryResolver } = {}) {
+    const resolvers = {}
+
+    const cache = new InMemoryCache({
+      typePolicies: {
+        Query: {
+          fields: {
+            word: {
+              read: wordQueryResolver
+            }
+          }
+        }
+      }
+    })
+
+    this._gql = new ApolloClient({
+      cache,
+      link: new HttpLink({ uri: '/graphql', fetch }),
```

Comment:
Here `fetch` is a reference to the `fetch` API of the browser: https://www.apollographql.com/docs/link/links/http/#fetch-polyfill. But that creates an issue with testing in node/Jest environment. We need to use a polyfill. The one I've used, `cross-fetch` has some issues that I was not able to resolve so far.

The option to provide a fetch-compatible API instead of using the only one provided by Apollo is part of a flexible Apollo link architecture.